quickshop-bukkit/src/main/java/com/ghostchu/quickshop/menu/ShopKeeperMenu.java

```diff
@@ -34,7 +36,9 @@ public class ShopKeeperMenu extends QuickShopMenu {
 
   public ShopKeeperMenu() {
 
-    this.rows = 3;
+    // Load rows from config or use default (4 rows for modern layout)
+    final GuiConfig.MenuConfig menuConfig = QuickShop.getInstance().getGuiConfig().getMenuConfig("keeper");
+    this.rows = menuConfig != null ? menuConfig.getRows() : 4;
```

Comment:
The ternary condition should have parentheses around it, with the question mark operator having no space before it.